os/mm/mm_heap/mm_getsize.c

```diff
@@ -0,0 +1,85 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+#include <sys/types.h>
+#include <tinyara/mm/mm.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static size_t mm_find_largest_freesize(struct mm_heap_s *heap)
+{
+	size_t largest_size = 0;
+	struct mm_freenode_s *fnode;
+	int i = 0;
+
+	for (i = 0; i < MM_NNODES; ++i) {
+		fnode = heap->mm_nodelist[i].flink;
+		if (fnode && largest_size < fnode->size) {
+			largest_size = fnode->size;
+		}
+	}
```

Comment:
Could you check my previous comment? Do we need to search entire node?

Reply:
If you change the code as I said, please leave comments for that.

Reply:
This is not to traverse the entire node, it only traverses the head of each nodelist.

Reply:
Oh if I start with end index, and found the node, then I can decide the largest.
I will fix this.
Thanks.

Reply:
> This way looks full searching of free node. Is't it possible to check first meet node with descending order?

Please find `descending order`. I think if you check it with descending order, you don't need to check next node. Because first meet is largest. Could you check it?

Reply:
Modified. Thanks.